tests/scripts/thread-cert/node.py

```diff
@@ -636,14 +597,11 @@ def get_eidcaches(self):
         eidcaches = []
         self.send_command('eidcache')
 
-        while True:
-            i = self._expect([r'([a-fA-F0-9\:]+) ([a-fA-F0-9]+)\r?\n', 'Done'])
-            if i == 0:
-                eid = self.pexpect.match.groups()[0].decode("utf-8")
-                rloc = self.pexpect.match.groups()[1].decode("utf-8")
-                eidcaches.append((eid, rloc))
-            elif i == 1:
-                break
+        pattern = self._prepare_pattern(r'([a-fA-F0-9\:]+) ([a-fA-F0-9]+)')
+        while self._expect(pattern):
```

Comment:
I think in long term we can eliminate all calls to `_expect`, leaving only `_expect_results` and other methods that are based on `_expect_results`.
For example, current calls to `self._expect('Done')` are not strictly correct. It is just expecting string `Done`, however the correct behavior is to expect a line of `Done`. 
Thoughts.

Reply:
Agree. Maybe replacing send_command with execute_command and let it return expected results is even cleaner. But I think it's beyond the purpose of this PR.